File: DynamicKey/AgoraDynamicKey/python3/sample/RtmTokenBuilderSample.py
#! /usr/bin/python
# ! -*- coding: utf-8 -*-

import sys
import os
import time
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from src.RtmTokenBuilder import RtmTokenBuilder,Role_Rtm_User

appID = "b1a776384fe24b58a43030c834b8f7dd"
appCertificate = "5b4b7878c61740dcb5a35ae72f0fefdd"
user = "1234"
expirationTimeInSeconds = 3600
currentTimestamp = int(time.time())
privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds

def main():
    token = RtmTokenBuilder.buildToken(appID, appCertificate, user, Role_Rtm_User, privilegeExpiredTs)
    print("Rtm Token: {}".format(token))


if __name__ == "__main__":
    main()

File: DynamicKey/AgoraDynamicKey/nodejs/sample/RtmTokenBuilderSample.js
const RtmTokenBuilder = require('../src/RtmTokenBuilder').RtmTokenBuilder;
const RtmRole = require('../src/RtmTokenBuilder').Role;
const Priviledges = require('../src/AccessToken').priviledges;
const appID  = "b1a776384fe24b58a43030c834b8f7dd";
const appCertificate = "5b4b7878c61740dcb5a35ae72f0fefdd";
const account = "1234";

const expirationTimeInSeconds = 3600
const currentTimestamp = Math.floor(Date.now() / 1000)

const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds

const token = RtmTokenBuilder.buildToken(appID, appCertificate, account, RtmRole, privilegeExpiredTs);
console.log("Rtm Token: " + token);
